src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -415,7 +415,7 @@ where
 
             // We just use an advancing number for our capability. No one cares
             // about what this actually is, downstream.
-            let mut batch_counter = 0;
+            let mut batch_counter = timely::progress::Timestamp::minimum();
```

Comment:
This one was expressly not a "timestamp", because we're just counting things. The `checked_add()` below seems wise, but we could stick to `u64` here.

Reply:
I'm of the opposite opinion! It seems to me that `batch_counter` is a timestamp (whose meaning is different fro other timestamps, but neverthless a timestamp) masquerading as an int because it is used in `cap_set.delayed(&batch_counter);`, at which point it must be a `Timestamp` type. I think we have exactly two choices: 1) make `batch_counter` a `Timestamp` (this PR) or 2) keep it an int, and explicitly convert it to a `Timestamp` on each call to `delayed`.

Reply:
aha! It should be a timely `Timestamp` but not a materialize `Timestamp`. It might be that we can't easily keep them separate here because the scope `G` comes with a "Timestamp" that is used for `Stream`s created in it. (Though I think there are ways around that).

So the change seems fine for now! Sorry for the confusion.